### ISSUE TITLE
Cannot read property 'slug' of undefined

### ISSUE THREAD

**Issue by mikemand** [cnvs/canvas#650]
I'm currently using the `develop` branch (4cfe89a593cd9e2691bd91bc6268ce77091549c4 is what my composer.lock says) because my App is on Laravel 7 already.

When I try to add a custom Featured Image to my post (either before it is created, or after it has been saved as a draft, or even after it has been published), nothing happens. The image is showing in the upload area, and it says upload complete. This does not happen when I use Unsplash. I've made sure I have the latest assets published.

(By the way, 👍 for Filepond - I'm using it in another project and it is great.)

In the browser's console I have this error:

```
TypeError: Cannot read property 'slug' of undefined
    at a.<anonymous> (app.js?id=4f95e6a5db98bc9497d2:2)
    at a.INkZ.e._render (app.js?id=4f95e6a5db98bc9497d2:2)
    at a.r (app.js?id=4f95e6a5db98bc9497d2:2)
    at fn.INkZ.fn.get (app.js?id=4f95e6a5db98bc9497d2:2)
    at fn.INkZ.fn.run (app.js?id=4f95e6a5db98bc9497d2:2)
    at un (app.js?id=4f95e6a5db98bc9497d2:2)
    at Array.<anonymous> (app.js?id=4f95e6a5db98bc9497d2:2)
    at Ke (app.js?id=4f95e6a5db98bc9497d2:2)
```

And expanded:

```
Fe @ app.js?id=4f95e6a…:2
Xe @ app.js?id=4f95e6a…:2
We @ app.js?id=4f95e6a…:2
INkZ.e._render @ app.js?id=4f95e6a…:2
r @ app.js?id=4f95e6a…:2
INkZ.fn.get @ app.js?id=4f95e6a…:2
INkZ.fn.run @ app.js?id=4f95e6a…:2
un @ app.js?id=4f95e6a…:2
(anonymous) @ app.js?id=4f95e6a…:2
Ke @ app.js?id=4f95e6a…:2
Promise.then (async)
Ue @ app.js?id=4f95e6a…:2
tt @ app.js?id=4f95e6a…:2
(anonymous) @ app.js?id=4f95e6a…:2
INkZ.fn.update @ app.js?id=4f95e6a…:2
INkZ.de.notify @ app.js?id=4f95e6a…:2
set @ app.js?id=4f95e6a…:2
(anonymous) @ app.js?id=4f95e6a…:2
Promise.catch (async)
saveActivePost @ app.js?id=4f95e6a…:2
(anonymous) @ app.js?id=4f95e6a…:2
(anonymous) @ app.js?id=4f95e6a…:2
(anonymous) @ app.js?id=4f95e6a…:2
L2JU.u._withCommit @ app.js?id=4f95e6a…:2
L2JU.u.commit @ app.js?id=4f95e6a…:2
commit @ app.js?id=4f95e6a…:2
saveActivePost @ app.js?id=4f95e6a…:2
(anonymous) @ app.js?id=4f95e6a…:2
L2JU.u.dispatch @ app.js?id=4f95e6a…:2
dispatch @ app.js?id=4f95e6a…:2
save @ app.js?id=4f95e6a…:2
clickDone @ app.js?id=4f95e6a…:2
He @ app.js?id=4f95e6a…:2
n @ app.js?id=4f95e6a…:2
Gr.o._wrapper @ app.js?id=4f95e6a…:2
```

**Comment by mikemand** [comment]
Fresh install. I installed it a few days ago while I was still on Laravel 6, then upgraded my app to Laravel 7.

I'm using S3 as my storage, here are my environment variables:

```
CANVAS_STORAGE_DISK=s3
CANVAS_STORAGE_PATH=blog
```

I know my connection to S3 is working because I also use Spatie's media library package, but maybe it didn't create the path?

**Comment by mikemand** [comment]
I updated to your latest commit (b2e5ded3dce6a05c0bad75af653e3667d53d7cae), and have a better error message now:

```
[Vue warn]: Error in render: "TypeError: Cannot read property 'slug' of undefined"

found in

---> <SettingsModal> at resources/js/components/modals/SettingsModal.vue
       <PostsEdit> at resources/js/screens/posts/PostEdit.vue
         <Root>
```

**Comment by mikemand** [comment]
Aha! I found the problem. Telescope Toolbar is adding its assets to the file path that is getting saved to the `featured_image` column.

Something I've noticed is that Telescope Toolbar (and also DebugBar, I think) add their assets to any response that is not JSON or plain text. So you'll need to set the content type before this return: https://github.com/cnvs/canvas/blob/b2e5ded3dce6a05c0bad75af653e3667d53d7cae/src/Http/Controllers/MediaController.php#L28

**Comment by mikemand** [comment]
Turns out Telescope Toolbar doesn't have this ability. I've opened an issue here: https://github.com/fruitcake/laravel-telescope-toolbar/issues/25

We can close this if you want since it isn't technically relevant to Canvas. A quick fix would be to add `header('Content-Type: text/plain')` before the return I mentioned above, but I don't know if it is fair to ask you to change Canvas for some obscure issue from another (development-only) package.
